src/main/java/life/qbic/App.java

```diff
@@ -37,12 +38,34 @@ public static void main(String[] args) throws IOException {
    */
   private static QbicDataDownloader loginToOpenBIS(
       PostmanCommandLineOptions commandLineParameters) {
-    LOG.info(String.format("Please provide password for user '%s':", commandLineParameters.user));
 
-    String password = OpenBISPasswordParser.readPasswordFromConsole();
+    String password = "";
+
+    //Password read in from environment Variable
+    if ((!(commandLineParameters.environmentVariableName.isEmpty()))) {
+
+      if (System.getenv(commandLineParameters.environmentVariableName) == null) {
+
+        LOG.info(String.format("Unfortunately, the given environment variable does not exist. " +
+                "Please provide password for user '%s':", commandLineParameters.user));
+
+        password = OpenBISPasswordParser.readPasswordFromConsole();
+      }
+      else {
+        password = System.getenv(commandLineParameters.environmentVariableName);
+      }
+    }
+    
+    //Password read in from Console
```

Comment:
A method can be extracted here ( for example `readPasswordFromEnvVariable(String variableName)`) This would replace your comment `//Password read in from environment Variable` while increasing the readability of the code

Reply:
As the method `readPasswordFromConsole` is very explicit, I think this comment can be removed.